### PR TITLE
[feat/VEGA-5285] Multiselect possibility has been added to vega-ui

### PR DESCRIPTION
**Pull request by Alexandr1111**
## Задача
Cсылка в Jira GPN на задачу: https://jira.konakov.tv/browse/VEGA-5285

## Problem statement
multiselect possibility has been added to vega-ui.

https://artcpt.atlassian.net/browse/VEGA-5285
## Solution details

1. current row selected by click on "**ID**" column
2. **ctrl** click allows to select several wors
3. **shift** click allows to select all rows between initially checked row and target row

## Type
- [x] Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control
- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [x] PR: Assignee chosen and necessary labels are set
- [x] PR: Current form is filled out (where it makes sense)
- [x] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [the rules](../docs/commits-style.md)
- [x] JS: There's no errors/warnings in the browser dev console
- [x] Layout: Uses [variables](https://github.com/gpn-prototypes/ui-kit/tree/master/src/components/Theme) and [CSS classes](https://github.com/gpn-prototypes/ui-kit/blob/master/src/utils/whitepaper/whitepaper.css) from [GPN design system](https://github.com/gpn-prototypes/ui-kit)
- [ ] Layout: Tested with various content amounts
- [ ] Docs: All changes to the component API are reflected and key points are described
- [ ] Storybook: Stories are written/updated for components
- [ ] Storybook: Stories follow [the rules](https://github.com/gpn-prototypes/vega-ui/blob/master/docs/storybook.md)
- [ ] Tests: New features and bugfixes are covered with tests
- [x] Tests: Existing tests passed successfully

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

File: src/components/vega-table/vega-row/VegaRow.css [hunk 1]
```diff
@@ -24,6 +24,9 @@
   &.is-selected {
     z-index: 10;
     contain: none;
+    background-color: rgb(75 81 84);
```
Comment (SergeyKatugin) [inline src/components/vega-table/vega-row/VegaRow.css:27]: Из константы?

File: src/components/vega-table/VegaTable.stories.tsx [hunk 2]
```diff
@@ -61,6 +64,7 @@ storiesOf('ui/VegaTable', module)
               !!(column?.key === '2544f6ae-5950-45f5-b81e-197ae6f1765d'),
           }}
           headerRowHeight={50}
+          isRowSelected={selectedRows}
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx:67]: Переделано на 'selectedRows'

File: src/components/vega-table/cells/DropDownCell/DropDownCell.tsx [hunk 3]
```diff
@@ -61,6 +61,7 @@ const DropDownCell = <T extends GridRow = DefaultCellData>(
   };
 
   const handleCellClick = () => {
+    alert('tyjt');
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/cells/DropDownCell/DropDownCell.tsx:64]: Убрано

File: src/components/vega-table/filter-popover/TableFilterPopover.tsx [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/filter-popover/TableFilterPopover.tsx:2]: Убрано

File: src/components/vega-table/VegaTable.stories.tsx [hunk 5]
```diff
@@ -95,6 +99,72 @@ storiesOf('ui/VegaTable', module)
             // eslint-disable-next-line no-console
             console.log('onHeaderContext', event, ref, columnKey);
           }}
+          setCtrl={(isActive: boolean) => setIsCtrl(isActive)}
+          setShift={(isActive: boolean) => setIsShift(isActive)}
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx]: Переделано, это в пределах idCell теперь хранится и обрабатывается

File: src/components/vega-table/VegaTable.stories.tsx [hunk 6]
```diff
@@ -95,6 +99,72 @@ storiesOf('ui/VegaTable', module)
...
+                    i <= newLastSelectedRow;
+                    i += 1
+                  ) {
+                    resultRows.push(i);
+                  }
+                } else {
+                  for (let i = firstSelectedRow; i <= lastSelectedRow; i += 1) {
+                    resultRows.push(i);
+                  }
+                }
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx]: Переделано, это в пределах idCell теперь хранится и обрабатывается

File: src/components/vega-table/cells/DelimeterCell.tsx [hunk 7]
```diff
@@ -15,7 +15,7 @@ const DelimeterCell = (props: Props) => {
   return (
     <div
       aria-colindex={column.idx + 1}
-      data-test-id="delimeter-cell"
+      data-testid="delimeter-cell"
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/cells/DelimeterCell.tsx:18]: Проверил, чёт в e2eхах вообще нет этого delimeter-cell нигде

File: src/components/vega-table/cells/IdCell.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    <div
+      aria-colindex={column.idx + 1}
+      aria-selected={isCellSelected}
+      data-testid="id-cell"
+      ref={combinedRef}
+      className={cn('rdg-cell', column.cellClass, {
+        'rdg-cell-frozen': column.frozen,
+        'rdg-cell-dragged-over': isDraggedOver,
+        'rdg-cell-copied': isCopied,
+        'rdg-cell-disabled': isCellDisabled,
```
[the remark this answers is not in the document]
Comment (Alexandr1111) [reply src/components/vega-table/cells/IdCell.tsx]: подчистил дисейблед, точно не нужен тут

File: src/components/vega-table/vega-row/VegaRow.css (hunk 1 above)
Comment (Alexandr1111) [reply src/components/vega-table/vega-row/VegaRow.css:27]: Убрал это свойство, потому как не было различимо к запасам или к ресурсам относится строка

File: reports/junit.xml [hunk 9]
```diff
@@ -0,0 +1,545 @@
...
+    <testcase classname="Tree drag and drop работает, при событии drop обработчик onPaste срабатывает и передает нужные параметры" name="Tree drag and drop работает, при событии drop обработчик onPaste срабатывает и передает нужные параметры" time="0.021">
+    </testcase>
+    <testcase classname="Tree отображаются чекбоксы" name="Tree отображаются чекбоксы" time="0.011">
+    </testcase>
+    <testcase classname="Tree убирается чекбоксы через withCheckElementSwitcher" name="Tree убирается чекбоксы через withCheckElementSwitcher" time="0.006">
+    </testcase>
+    <testcase classname="Tree проверяем отображение заранее выбранных чекбоксов" name="Tree проверяем отображение заранее выбранных чекбоксов" time="0.018">
+    </testcase>
+  </testsuite>
+</testsuites>
```
Comment (ndrzhn) [inline reports/junit.xml:545]: этот файл не нужен, удалю

File: src/components/vega-table/cells/IdCell.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+  const combinedRef = useCombinedRefs(ref, innerRef);
+  const [isCtrl, setIsCtrl] = useState<boolean>(false);
+  const [isShift, setIsShift] = useState<boolean>(false);
+
+  const position = {
+    idx: column.idx,
+    rowIdx: currentRowIdx,
+  };
+
+  const handlerKeyDown = useCallback((e) => {
```
Comment (ndrzhn) [inline src/components/vega-table/cells/IdCell.tsx:51]: мне кажется, лучше назвать handleKeyDown, и ниже тоже

**Review by ndrzhn — APPROVED** [review]

**Review by AllPein — APPROVED** [review]

File: src/components/vega-table/VegaTable.tsx [hunk 11]
```diff
@@ -39,6 +40,9 @@ export const VegaTable = <
     handleColumnsReorder,
     sorting,
     setSorting,
+    selectRow,
+    setCtrl,
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/vega-table/VegaTable.tsx]: это что-то устаревшее

File: src/components/vega-table/VegaTable.tsx [hunk 12]
```diff
@@ -39,6 +40,9 @@ export const VegaTable = <
...
+    setShift,
```
[the remark this answers is not in the document]
Comment (ndrzhn) [reply src/components/vega-table/VegaTable.tsx]: и это устаревшее

**Pull request merged**